newIDE/app/src/Profile/ForgotPasswordDialog.js

```diff
@@ -0,0 +1,87 @@
+// @flow
+import { Trans } from '@lingui/macro';
+
+import React from 'react';
+import FlatButton from '../UI/FlatButton';
+import Dialog, { DialogPrimaryButton } from '../UI/Dialog';
+import { Column } from '../UI/Grid';
+import TextField from '../UI/TextField';
+import { type ForgotPasswordForm } from '../Utils/GDevelopServices/Authentication';
+import LeftLoader from '../UI/LeftLoader';
+import Text from '../UI/Text';
+
+type Props = {|
+  onClose: () => void,
+  onForgotPassword: ForgotPasswordForm => Promise<void>,
+|};
+
+const ForgotPasswordDialog = ({ onClose, onForgotPassword }: Props) => {
+  const [email, setEmail] = React.useState('');
+  const [resetDone, setResetDone] = React.useState(false);
+  const [resetInProgress, setResetInProgress] = React.useState(false);
+
+  const doResetPassword = async () => {
+    if (resetInProgress || !email) return;
+    setResetInProgress(true);
+
+    await onForgotPassword({
```

Comment:
Should you catch an error here?

Reply:
The error is already caught by the function itself, and not shown to the user;
The idea is to prevent giving any hint to a potential attacker if the email is in our DB